File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int res = INT_MAX,
            l = 0,
            r = 0,
            t = 0,
            n = nums.size();

        for (r; r < n; ++r) {
            t += nums[r];
            while (t >= target) {
                res = min(res, r - l + 1);
                t -= nums[l];
                ++l;
            } 
        }

        // summation of entire array was smaller than target
        if (res == INT_MAX) return 0;
        return res;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.rs
use std::cmp::min;

impl Solution {
    pub fn min_sub_array_len(target: i32, nums: Vec<i32>) -> i32 {
        let mut l:usize = 0;
        let (mut sum, mut res) = (0, i32::MAX);

        for r in 0..nums.len() {
            sum += nums[r];
            while sum >= target {
                l += 1;
                sum -= nums[l];
                res = min(res, (r - l + 1) as i32);    
            }
        }

        if res == i32::MAX {
            0
        } else {
            res
        }
    }
}